File: solutions/cpp/log-levels/1/log_levels.cpp
#include <string>

namespace log_line {
std::string message(std::string line) {
    // Find the colon and skip the colon and space
    size_t colon_pos = line.find(':');
    // The message starts after the colon and space
    return line.substr(colon_pos + 2);
}

std::string log_level(std::string line) {
    // Find the opening and closing brackets
    size_t bracket_start = line.find('[');
    size_t bracket_end = line.find(']');
    // Extract the level (excluding the brackets)
    return line.substr(bracket_start + 1, bracket_end - bracket_start - 1);
}

std::string reformat(std::string line) {
    // Get the message and log level using the existing functions
    std::string msg = message(line);
    std::string level = log_level(line);
    // Return the reformatted string
    return msg + " (" + level + ")";
}
}  // namespace log_line

File: solutions/java/log-levels/1/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
    String[] tokens = logLine.split(":\\s*", 2); 
    return tokens[1].trim();
}

    public static String logLevel(String logLine) {
    String[] tokens = logLine.split(":\\s*", 2);
    return tokens[0].replace("[", "").replace("]", "").trim().toLowerCase();
}

    public static String reformat(String logLine) {
      String message = message(logLine);
      String level = logLevel(logLine);
      return message + " ("+ level + ")";
    }
}
